### ISSUE TITLE
`no-useless-undefined` triggers incorrectly with set.delete(undefined)

### ISSUE THREAD

**Issue by mkahvi** [sindresorhus/eslint-plugin-unicorn#2737]
`set.delete(undefined)` incorrectly is detected as useless even though a set can contain undefined.

```js
const set = new Set([1,2,undefined]);
set.delete(undefined); // rule warning here
```
